service/src/main/java/org/broadinstitute/listener/relay/transport/DefaultTargetResolver.java

```diff
@@ -59,6 +56,23 @@ public URL createTargetUrl(@NonNull URI relayedRequestUri) throws InvalidRelayTa
         relayedRequestUri, properties.getTargetProperties().isRemoveEntityPathFromHttpUrl());
   }
 
+  private String resolveTargetHostUsingRules(@NonNull URI relayedRequestUri) {
+    List<TargetRoutingRule> rules = properties.getTargetProperties().getTargetRoutingRules();
+    if (rules == null || rules.size() == 0) {
+      return defaultTargetHost;
+    }
+    Optional<String> ruleTargetHost =
+        rules.stream()
+            .filter(r -> relayedRequestUri.toString().contains(r.pathContains()))
+            .map(r -> r.targetHost())
+            .findFirst();
+
+    if (ruleTargetHost.isPresent()) {
+      return ruleTargetHost.get();
+    }
+    return defaultTargetHost;
```

Comment:
In most cases you shouldn't need to call `isPresent()` followed by `get()`.  If you don't have any `else` behavior, use `map()`. If you have an `else` behavior, use one of the `orElse()` methods. In this case:
```suggestion
    return ruleTargetHost.orElse(defaultTargetHost);
```